app/src/main/cpp/skyline/soc/gm20b/engines/maxwell_dma.cpp

```diff
@@ -30,167 +31,241 @@ namespace skyline::soc::gm20b::engine {
+
+            if (registers.launchDma->remapEnable) [[unlikely]] {
+                // Remapped buffer clears
+                if ((registers.remapComponents->dstX == Registers::RemapComponents::Swizzle::ConstA) &&
+                    (registers.remapComponents->dstY == Registers::RemapComponents::Swizzle::ConstA) &&
+                    (registers.remapComponents->dstZ == Registers::RemapComponents::Swizzle::ConstA) &&
+                    (registers.remapComponents->dstW == Registers::RemapComponents::Swizzle::ConstA) &&
+                    (registers.remapComponents->ComponentSize() == 4)) {
+                    for (auto mapping : dstMappings)
+                        interconnect.Clear(mapping, *registers.remapConstA);
+                } else {
+                    Logger::Warn("Remapped DMA copies are unimplemented!");
+                }
+            } else {
+                if (srcMappings.size() != 1 || dstMappings.size() != 1) [[unlikely]]
+                    channelCtx.asCtx->gmmu.Copy(u64{*registers.offsetOut}, u64{*registers.offsetIn}, *registers.lineLengthIn);
+                else
+                    interconnect.Copy(dstMappings.front(), srcMappings.front());
             }
-            default:
-                break;
         }
     }
 
-    void MaxwellDma::CopyPitchToBlockLinear() {
-        if (registers.dstSurface->blockSize.Depth() > 1 || registers.dstSurface->depth > 1) {
-            Logger::Warn("3D DMA engine copies are unimplemented");
-            return;
-        }
-
-        if (registers.dstSurface->blockSize.Width() != 1) {
-            Logger::Warn("DMA engine copies with block widths other than 1 are unimplemented");
-            return;
-        }
+    void MaxwellDma::HandleCopy(TranslatedAddressRange srcMappings, TranslatedAddressRange dstMappings, size_t srcSize, size_t dstSize, auto copyCallback) {
+        bool isSrcSplit{};
+        u8 *src{srcMappings.front().data()}, *dst{dstMappings.front().data()};
+        if (srcMappings.size() != 1) {
+            if (copyCache.size() < srcSize)
+                copyCache.resize(srcSize);
 
-        u32 bytesPerPixel{static_cast<u32>(registers.remapComponents->ComponentSize() * registers.remapComponents->NumSrcComponents())};
-        if (bytesPerPixel * *registers.lineLengthIn != *registers.pitchIn) {
-            Logger::Warn("Non-linear DMA source textures are not implemented");
-            return;
-        }
+            src = copyCache.data();
+            channelCtx.asCtx->gmmu.Read(src, u64{*registers.offsetIn}, srcSize);
 
-        if (registers.dstSurface->origin.x || registers.dstSurface->origin.y) {
-            Logger::Warn("Non-zero origin DMA copies are not implemented");
-            return;
+            isSrcSplit = true;
         }
+        if (dstMappings.size() != 1) {
+            // If both the source and destination are split
```

Comment:
You can avoid duplicating resize logic here by doing like
size_t offset = srcSplit ? srcSize : 0 
then following the split path